cmd/upgrade-metadata.go

```diff
@@ -186,6 +187,121 @@ func getWriteFiles(networks sls_common.NetworkArray, ipamNetworks bss.CloudInitI
+		if err != nil {
+			log.Fatalf("Failed to decode raw network extra properties to correct structure: %s", err)
+		}
+
+		networkEPs[network.Name] = &networkExtraProperties
+	}
+
+	var globalHostRecords bss.HostRecords
+
+	// Add in the NCN Interfaces
+	for _, managementNCN := range managementNCNs {
+		var ncnExtraProperties sls_common.ComptypeNode
+		err := mapstructure.Decode(managementNCN.ExtraPropertiesRaw, &ncnExtraProperties)
+		if err != nil {
+			log.Fatalf("Failed to decode raw NCN extra properties to correct structure: %s", err)
+		}
+
+		if len(ncnExtraProperties.Aliases) == 0 {
+			log.Fatalf("NCN has no aliases defined in SLS: %+v", managementNCN)
+		}
+
+		ncnAlias := ncnExtraProperties.Aliases[0]
+
+		// Add in the NCN interface host records
+		ipamNetworks := getIPAMForNCN(managementNCN, networks, "chn")
+		for network, ipam := range ipamNetworks {
+			// Get the IP of the NCN for this network.
+			ip, _, err := net.ParseCIDR(ipam.CIDR)
+			if err != nil {
+				log.Fatalf("Failed to parse BSS IPAM Network CIDR (%s): %s", ipam.CIDR, err)
+			}
+
+			hostRecord := bss.HostRecord{
+				IP:      ip.String(),
+				Aliases: []string{fmt.Sprintf("%s.%s", ncnAlias, network)},
+			}
+
+			// The NMN network gets the privledge of also containing the bare NCN Alias without network domain.
+			if strings.ToLower(network) == "nmn" {
+				hostRecord.Aliases = append(hostRecord.Aliases, ncnAlias)
+			}
+			globalHostRecords = append(globalHostRecords, hostRecord)
+		}
+
+		// Next add the NCN BMC host record
+		bmcXname := base.GetHMSCompParent(managementNCN.Xname)
+		globalHostRecords = append(globalHostRecords,
+			buildBSSHostRecords(networkEPs, "HMN", "bootstrap_dhcp", bmcXname, []string{fmt.Sprintf("%s-mgmt", ncnAlias)}),
+		)
+	}
+
+	// Add kubeapi-vip
+	globalHostRecords = append(globalHostRecords,
+		buildBSSHostRecords(networkEPs, "NMN", "bootstrap_dhcp", "kubeapi-vip", []string{"kubeapi-vip", "kubeapi-vip.nmn"}),
+	)
+
+	// Add rgw-vip
+	globalHostRecords = append(globalHostRecords,
+		buildBSSHostRecords(networkEPs, "NMN", "bootstrap_dhcp", "rgw-vip", []string{"rgw-vip", "rgw-vip.nmn"}),
+	)
+
+	// Using the original InstallNCN as the host for pit.nmn
+	// HACK, I'm assuming ncn-m001
+	globalHostRecords = append(globalHostRecords,
+		buildBSSHostRecords(networkEPs, "NMN", "bootstrap_dhcp", "ncn-m001", []string{"pit", "pit.nmn"}),
+	)
+
+	// Add in packages.local and registry.local pointing toward the API Gateway
+	globalHostRecords = append(globalHostRecords,
+		buildBSSHostRecords(networkEPs, "NMNLB", "nmn_metallb_address_pool", "istio-ingressgateway", []string{"packages.local", "registry.local"}),
+	)
+
+	// Add entries for switches
+	nmnNetSubnet, err := networkEPs["NMN"].LookupSubnet("network_hardware")
+	if err != nil {
+		log.Fatal("Unable to find network_hardware in the NMN network")
+	}
+
+	for _, ipResveration := range nmnNetSubnet.IPReservations {
```

Comment:
Just a typo but doesn't affect the functionality since it carries through:
   ipResveration -> ipReservation

Reply:
Fixed